lib/swoosh/adapters/mailgun.ex

```diff
@@ -79,19 +79,23 @@ defmodule Swoosh.Adapters.Mailgun do
   end
   defp prepare_custom_vars(body, _email), do: body
 
+  defp prepare_attachments(body, %{attachments: []}), do: body
+  defp prepare_attachments(body, %{attachments: attachments}) do
+    Map.put(body, :attachment, Enum.map(attachments, &File.read!(&1.path)))
```

Comment:
You are right. I'm in the middle of testing it and fixing the implementation. It's been too long since I had this code, I forgot whether I tested it already. I've added a todo in the PR description.